src/Foliage.rb

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env ruby
+
+class Foliage
+  @height
+  @offset
+
+  def initialize
+    @height = 0
+    @offset = 0
+  end
+
+  # Increases the height of the foliage by one
+  def grow
+    @height += 1
+    return self
+  end
+
+  # Sets the height of the foliage
+  # Height must not be negative
+  # Raises RangeError if the height is negative
+  # Returns self, the current Foliage instance
+  def set_height(height)
+    if (height < 0)
+      raise RangeError, "Height of the foliage cannot be negative."
+    end
+    @height = height
+    return self
+  end
+
+  # Returns the height of the foliage
+  def get_height
+    return @height
+  end
+
+  # Sets the offset of the foliage, in number of spaces
+  # Offset must not be negative
+  # Raises RangeError if the offset is negative
+  # Returns self, the current Foliage instance
+  def set_offset(offset)
+    if (offset < 0)
+      raise RangeError, "Offset of the foliage cannot be negative"
+    end
+    @offset = offset
+    return self
+  end
+
+  # Returns the offset of the foliage, in number of spaces
+  def get_offset
+    return @offset
+  end
+
+  # Draws the foliage that is @height tall, with @offset offset
+  # Returns the string representing the foliage
+  # If the height is not positive, returns a single "leaf"
+  def draw
+    if (@height < 0)
+      raise RangeError, "Height of the foliage cannot be negative."
```

Comment:
Need to document error

Reply:
https://github.com/Wall-Dough/pointless/pull/5/commits/366197a23b31aafb43765ce4e30a9deb169dfa1a